src/state/store.ts

```diff
@@ -25,202 +21,213 @@ export interface UIState {
-            if (!str) return undefined;
-            try {
-                const obj = JSON.parse(str);
-                const v = lodashGet(obj, path);
-                return typeof v === "string" || typeof v === "number" ? v : undefined;
-            } catch {
-                console.warn("Failed to retrieve value.")
-                return undefined;
-            }
+            const rec = recipes[selectedRecipeId]?.defaultRecipeData;
+            if (!rec) return undefined;
+            const v = lodashGet(rec, path);
+            return (typeof v === "string" || typeof v === "number") ? v : undefined;
         },
 
+        startPacking: async () => {
+            const { selectedRecipeId, recipes } = get();
+            const rec = recipes[selectedRecipeId];
+            if (!rec) return;
 
+            const recipeString = jsonToString(buildCurrentRecipeObject(rec))
+
+            set({ isPacking: true, packingData: EMPTY_PACKING_DATA });
+
+            const startedAtMs = Date.now();
 
-        startPacking: async (callback) => {
-            const s = get();
-            const input = s.inputOptions[s.selectedInputName];
-            const configId = input?.config ?? "";
-            const recipeString = s.recipes[s.selectedRecipeId]?.currentString ?? "";
-            set({ isPacking: true });
             try {
-                await callback(s.selectedRecipeId, configId, recipeString);
+                const { response, data } = await submitJob(selectedRecipeId, recipeString, rec.configId);
+                set((state) => ({ packingData: { ...state.packingData, jobStatus: JOB_STATUS.SUBMITTED } }));
+
+                if (!response.ok) {
+                    set((state) => ({ packingData: { ...state.packingData, jobStatus: JOB_STATUS.FAILED, jobLogs: JSON.stringify(data) } }));
+                    return;
+                }
+
+                const newJobId: string = data.jobId;
+                set(state => ({
+                    packingData: {
+                        ...state.packingData,
+                        jobId: newJobId,
+                        jobStatus: JOB_STATUS.STARTING,
+                    }
+                }));
+
+                const finalStatus = await pollForJobStatus(newJobId, (next) =>
+                    set(state => ({ packingData: { ...state.packingData, jobStatus: next } }))
+                );
+
+                const runTime = (Date.now() - startedAtMs) / 1000;
+
+                if (finalStatus.status === JOB_STATUS.DONE) {
+                    const url = await buildResultUrl(newJobId);
+                    const outputDir = await getOutputsDirectory(newJobId);
+                    set(state => ({
+                        packingData: {
+                            ...state.packingData,
+                            jobStatus: JOB_STATUS.DONE,
+                            resultUrl: url,
+                            outputDir,
+                            runTime,
```

Comment:
without jobstatus being here, this object makes sense as `results` or whatever. I wonder if jobstatus can be in the `LOADING`, `PACKING` enum. so the full enum is `LOADING` + all the job status options. or maybe that wouldnt be helpful